OK, I am stuck, tbh.

The build fails differently for x86 and x64 platforms. On x86 I am seeing:

```
make[5]: Entering directory '/c/vcpkg/buildtrees/libtasn1/x86-windows-dbg/lib/gl'
Makefile:721: update target 'c-ctype.lo' due to: ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c
source='../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c' object='c-ctype.lo' libtool=yes \
DEPDIR=.deps depmode=msvc7 /bin/sh ../.././../src/4.16.0-e9554bb491.clean/build-aux/depcomp \
/bin/sh ../../libtool  --tag=CC   --mode=compile /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../..   -DWIN32 -D_WINDOWS -D_DEBUG  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -c -o c-ctype.lo ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c
Makefile:721: update target 'strverscmp.lo' due to: ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c
source='../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c' object='strverscmp.lo' libtool=yes \
DEPDIR=.deps depmode=msvc7 /bin/sh ../.././../src/4.16.0-e9554bb491.clean/build-aux/depcomp \
/bin/sh ../../libtool  --tag=CC   --mode=compile /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../..   -DWIN32 -D_WINDOWS -D_DEBUG  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -c -o strverscmp.lo ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c
libtool: compile:  /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../.. -DWIN32 -D_WINDOWS -D_DEBUG -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -RTC1 -Od -c -showIncludes ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c  -DDLL_EXPORT -DPIC -o .libs/c-ctype.obj
c-ctype.c
libtool: compile:  /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../.. -DWIN32 -D_WINDOWS -D_DEBUG -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -RTC1 -Od -c -showIncludes ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c  -DDLL_EXPORT -DPIC -o .libs/strverscmp.obj
strverscmp.c
Makefile:687: update target 'libgnu.la' due to: c-ctype.lo strverscmp.lo
/bin/sh ../../libtool  --tag=CC   --mode=link /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1  -no-undefined -LC:/vcpkg/installed/x86-windows/debug/lib -LC:/vcpkg/installed/x86-windows/debug/lib/manual-link -o libgnu.la  c-ctype.lo strverscmp.lo -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -lcomdlg32 -ladvapi32 -lgettimeofday -lgetopt
libtool: link: lib.exe cru .libs/gnu.lib .libs/c-ctype.obj .libs/strverscmp.obj 
Microsoft (R) Library Manager Version 14.28.29337.0
Copyright (C) Microsoft Corporation.  All rights reserved.

LINK : fatal error LNK1181: cannot open input file 'cru'
make[5]: Leaving directory '/c/vcpkg/buildtrees/libtasn1/x86-windows-dbg/lib/gl'
```

And on x64:

```
make[5]: Entering directory '/c/vcpkg/buildtrees/libtasn1/x64-windows-dbg/lib/gl'
Makefile:721: update target 'c-ctype.lo' due to: ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c
source='../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c' object='c-ctype.lo' libtool=yes \
DEPDIR=.deps depmode=msvc7 /bin/sh ../.././../src/4.16.0-e9554bb491.clean/build-aux/depcomp \
/bin/sh ../../libtool  --tag=CC   --mode=compile /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../..   -DWIN32 -D_WINDOWS -D_DEBUG  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -c -o c-ctype.lo ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c
Makefile:721: update target 'strverscmp.lo' due to: ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c
source='../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c' object='strverscmp.lo' libtool=yes \
DEPDIR=.deps depmode=msvc7 /bin/sh ../.././../src/4.16.0-e9554bb491.clean/build-aux/depcomp \
/bin/sh ../../libtool  --tag=CC   --mode=compile /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../..   -DWIN32 -D_WINDOWS -D_DEBUG  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -c -o strverscmp.lo ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c
libtool: compile:  /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../.. -DWIN32 -D_WINDOWS -D_DEBUG -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -RTC1 -Od -c -showIncludes ../.././../src/4.16.0-e9554bb491.clean/lib/gl/strverscmp.c -o strverscmp.obj
strverscmp.c
libtool: compile:  /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe -DHAVE_CONFIG_H -I. -I../.././../src/4.16.0-e9554bb491.clean/lib/gl -I../.. -DWIN32 -D_WINDOWS -D_DEBUG -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -RTC1 -Od -c -showIncludes ../.././../src/4.16.0-e9554bb491.clean/lib/gl/c-ctype.c -o c-ctype.obj
c-ctype.c
Makefile:687: update target 'libgnu.la' due to: c-ctype.lo strverscmp.lo
/bin/sh ../../libtool  --tag=CC   --mode=link /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/compile cl.exe  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1  -no-undefined -LC:/vcpkg/installed/x64-windows/debug/lib -LC:/vcpkg/installed/x64-windows/debug/lib/manual-link -o libgnu.la  c-ctype.lo strverscmp.lo -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -lcomdlg32 -ladvapi32 -lgettimeofday -lgetopt
libtool: link: /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/ar-lib /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/ar-lib lib.exe cru .libs/libgnu.a  c-ctype.obj strverscmp.obj
make[5]: Leaving directory '/c/vcpkg/buildtrees/libtasn1/x64-windows-dbg/lib/gl'
```
With the error being:
`ar-lib: unknown action specified`

The x64 libtool call seems to reference `ar-lib` twice – could that explain  the "unknown action specified"?
`libtool: link: /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/ar-lib /c/vcpkg/buildtrees/libtasn1/src/4.16.0-e9554bb491.clean/build-aux/ar-lib lib.exe cru .libs/libgnu.a  c-ctype.obj strverscmp.obj`

--- a/ports/libtasn1/msvc_fixes.patch
+++ b/ports/libtasn1/msvc_fixes.patch
@@ -1,0 +1,86 @@
+diff --git a/src/asn1Coding.c b/src/asn1Coding.c
+index 86c1d07..6fcfc4d 100644
+--- a/src/asn1Coding.c
++++ b/src/asn1Coding.c
+@@ -23,7 +23,12 @@
+ #include <stdio.h>
+ #include <string.h>
+ #include <stdlib.h>
++#ifdef HAVE_UNISTD_H
+ #include <unistd.h>
++#endif
++#ifdef _MSC_VER
++#include <io.h>
++#endif
+ #include <getopt.h>
+ #include <assert.h>
+ 
+diff --git a/src/asn1Decoding.c b/src/asn1Decoding.c
+index c6f192e..3e4e21d 100644
+--- a/src/asn1Decoding.c
++++ b/src/asn1Decoding.c
+@@ -23,7 +23,12 @@
+ #include <stdio.h>
+ #include <string.h>
+ #include <stdlib.h>
++#ifdef HAVE_UNISTD_H
+ #include <unistd.h>
++#endif
++#ifdef _MSC_VER
++#include <io.h>
++#endif
+ #include <getopt.h>
+ #include <sys/types.h>
+ #include <sys/stat.h>
+diff --git a/src/asn1Parser.c b/src/asn1Parser.c
+index b6844a8..445c716 100644
+--- a/src/asn1Parser.c
++++ b/src/asn1Parser.c
+@@ -23,7 +23,12 @@
+ #include <stdio.h>
+ #include <string.h>
+ #include <stdlib.h>
++#ifdef HAVE_UNISTD_H
+ #include <unistd.h>
++#endif
++#ifdef _MSC_VER
++#include <io.h>
++#endif
+ #include <getopt.h>
+ #include <assert.h>
+ 
+diff --git a/src/benchmark.c b/src/benchmark.c
+index 010d58e..6c613df 100644
+--- a/src/benchmark.c
++++ b/src/benchmark.c
+@@ -21,9 +21,16 @@
+ #include <stdio.h>
+ #include <string.h>
+ #include <signal.h>
++#ifndef _MSC_VER
+ #include <sys/time.h>
++#endif
+ #include <time.h>
++#ifdef HAVE_UNISTD_H
+ #include <unistd.h>
++#endif
++#ifdef _MSC_VER
++#include <io.h>
++#endif
+ #include "benchmark.h"
+ 
+ int benchmark_must_finish = 0;
+diff --git a/src/benchmark.h b/src/benchmark.h
+index 3272649..6b6bf32 100644
+--- a/src/benchmark.h
++++ b/src/benchmark.h
+@@ -21,7 +21,9 @@
+ # define BENCHMARK_H
+ 
+ #include <config.h>
++#ifndef _MSC_VER
+ #include <sys/time.h>
++#endif
+ #include <time.h>
+ #include <signal.h>
+ #if defined _WIN32

--- a/ports/libtasn1/portfile.cmake
+++ b/ports/libtasn1/portfile.cmake
@@ -8,25 +8,32 @@ vcpkg_download_distfile(ARCHIVE
 
 vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
+    ARCHIVE ${ARCHIVE} 
     REF ${VERSION}
+    PATCHES
+        msvc_fixes.patch
 )
 
-# restore the default ac_cv_prog_cc_g flags, otherwise it fails to compile
-set(VCPKG_C_FLAGS "-g -O2")
-set(VCPKG_CXX_FLAGS "-g -O2")
+if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_MINGW)
+    # $LIBS is an environment variable that vcpkg already pre-populated with some libraries. 
+    # We need to re-purpose it when passing LIBS option to make to avoid overriding the vcpkg's own list.  
+    set(EXTRA_LIBS "LIBS=\"$LIBS -lgettimeofday -lgetopt\"") 
+else()
+    # restore the default ac_cv_prog_cc_g flags, otherwise it fails to compile
+set(VCPKG_C_FLAGS "-g -O2") 
+    set(VCPKG_CXX_FLAGS "-g -O2")
+endif()
 
 set(ENV{GTKDOCIZE} true)
 vcpkg_configure_make(
-    AUTOCONFIG
     SOURCE_PATH ${SOURCE_PATH}
     OPTIONS
         --disable-gtk-doc
+        "${EXTRA_LIBS}"
 )
 
 vcpkg_install_make()
 vcpkg_fixup_pkgconfig()
-vcpkg_copy_pdbs()
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/tools ${CURRENT_PACKAGES_DIR}/debug/share)
 file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

--- a/ports/libtasn1/portfile.cmake
+++ b/ports/libtasn1/portfile.cmake
@@ -8,7 +8,7 @@ vcpkg_download_distfile(ARCHIVE
 
 vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE} 
+    ARCHIVE ${ARCHIVE}
     REF ${VERSION}
     PATCHES
         msvc_fixes.patch
@@ -17,19 +17,31 @@ vcpkg_extract_source_archive_ex(
 if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_MINGW)
     # $LIBS is an environment variable that vcpkg already pre-populated with some libraries. 
     # We need to re-purpose it when passing LIBS option to make to avoid overriding the vcpkg's own list.  
-    set(EXTRA_LIBS "LIBS=\"$LIBS -lgettimeofday -lgetopt\"") 
+    set(EXTRA_OPTS "LIBS=\"$LIBS -lgettimeofday -lgetopt\"")
 else()
     # restore the default ac_cv_prog_cc_g flags, otherwise it fails to compile
-set(VCPKG_C_FLAGS "-g -O2") 
+    set(EXTRA_OPTS)
+    set(VCPKG_C_FLAGS "-g -O2") 
     set(VCPKG_CXX_FLAGS "-g -O2")
 endif()
 
+if(VCPKG_TARGET_IS_WINDOWS)
+    # These are hacks for MSVC since autoconf can't determine the absolute path correctly
+    foreach(H stdint limits string stddef)
+        find_file(H_PATH "${H}.h" PATHS $ENV{INCLUDE} NO_DEFAULT_PATH)
+        string(REPLACE "\\" "/" H_PATH "${H_PATH}")
+        list(APPEND EXTRA_OPTS "gl_cv_next_${H}_h='\"${H_PATH}\"'")
+    endforeach()
+endif()
+
 set(ENV{GTKDOCIZE} true)
+
 vcpkg_configure_make(
     SOURCE_PATH ${SOURCE_PATH}
     OPTIONS
         --disable-gtk-doc
-        "${EXTRA_LIBS}"
+        --disable-gcc-warnings
+        ${EXTRA_OPTS}
 )
 
 vcpkg_install_make()

--- a/ports/libtasn1/portfile.cmake
+++ b/ports/libtasn1/portfile.cmake
@@ -41,6 +41,7 @@ endif()
 set(ENV{GTKDOCIZE} true)
 
 vcpkg_configure_make(
+    USE_WRAPPERS
     SOURCE_PATH ${SOURCE_PATH}
     OPTIONS
         --disable-gtk-doc

--- a/ports/libtasn1/vcpkg.json
+++ b/ports/libtasn1/vcpkg.json
@@ -1,8 +1,18 @@
 {
   "name": "libtasn1",
   "version": "4.16.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "A secure communications library implementing the SSL, TLS and DTLS protocols",
   "homepage": "https://www.gnutls.org/",
-  "supports": "!windows"
+  "supports": "!uwp",
+  "dependencies": [
+    {
+      "name": "getopt",
+      "platform": "windows"
+    },
+    {
+      "name": "gettimeofday",
+      "platform": "windows"
+    }
+  ]
 }

